pkg/querier/http.go

```diff
@@ -98,9 +96,17 @@ func (q *QuerierAPI) RangeQueryHandler(w http.ResponseWriter, r *http.Request) {
 		serverutil.WriteError(err, w)
 		return
 	}
-	if err := marshal.WriteQueryResponseJSON(result, w); err != nil {
+
+	if r.Header.Get("Accept") == queryrange.ProtobufType {
+		w.Header().Add("Content-Type", queryrange.ProtobufType)
+		err = queryrange.WriteQueryResponseProtobuf(params, result, w)
+	} else {
+		w.Header().Add("Content-Type", queryrange.JSONType)
+		err = marshal.WriteQueryResponseJSON(result, w)
+	}
```

Comment:
should these be switch statements with a default case that logs an error?

Reply:
That would be a breaking change. If no `Accept` header is present we default to JSON.